src/Compilers/Core/CodeAnalysisTest/Diagnostics/OperationTestAnalyzer.cs

```diff
@@ -1303,4 +1304,93 @@ public sealed override void Initialize(AnalysisContext context)
                  OperationKind.EventAssignmentExpression);
         }
     }
+    
+    public class NullOperationSyntaxTestAnalyzer : DiagnosticAnalyzer
+    {
+        private const string ReliabilityCategory = "Reliability";
+
+        // We should not see this warning triggered by any code
+        public static readonly DiagnosticDescriptor NullOperationSyntaxDescriptor = new DiagnosticDescriptor(
+            "NullOperationSyntax",
+            "null operation Syntax found",
+            "An IOperation with Syntax property of value null is found",
```

Comment:
I couldn't create a simplified repro for the MVC template project, so have to came up with this to test if `Syntax == null`.